app/models/order_item.rb

```diff
@@ -38,6 +38,7 @@ def as_json(_options = {})
 
   def set_defaults
     self.state = "Created"
+    self.context = ManageIQ::API::Common::Request.current.to_h
```

Comment:
@mkanoor I'm assuming it's cheaper and simpler to save into `self.context` encoded.  But thought I would ask if you see any need to decode first?

Reply:
@syncrou It has to be stored as encoded and not any other way because we are going to fetch the context and send it back here to set the current context when we get feedback from Approval or Topology Service

https://github.com/ManageIQ/manageiq-api-common/blob/1bf945a647f86df3d9bd9552e288942ceff7e9db/lib/manageiq/api/common/request.rb#L27

Reply:
@syncrou It should be encoded